scripts/bb_regression.py

```diff
@@ -71,58 +106,118 @@ def main():
+        return model(crop_img.to(device))
+    evaluator = Evaluator(
+        Evaluator.Settings(period=opts.eval_period),
+        hub, model, test_loader, _eval_fn)
+
+    # TODO(Jiyong):
+    # All metrics evaluation should reset stats at eval_begin(),
+    # aggregate stats at eval_step(),
+    # and output stats at eval_end(). These signals are all implemented.
+    # What are the appropriate metrics to implement for bounding box regression?
+    def _on_eval_step(inputs, outputs):
+        pass
+    hub.subscribe(Topic.EVAL_STEP, _on_eval_step)
+
+    collect = Collect(hub, Topic.METRICS, [])
+
+    def _log_all(metrics: Dict[Topic, Any]):
+        pass
+    hub.subscribe(Topic.METRICS, _log_all)
+
+    orientation_loss_func = nn.L1Loss().to(device)
+    scale_loss_func = nn.L1Loss().to(device)
+
+    def _loss_fn(model: th.nn.Module, data):
         # Now that we're here, convert all inputs to the device.
-        data = {k: v for (k, v) in data.items()}
-
-        image = data['crop_img']
-        truth_orient = data[Schema.ORIENTATION]
-        truth_dim = data[Schema.SCALE]
-        # truth_conf = data['Confidence'].long().to(device)
-
-        # FIXME(Jiyong): parallelize by padding
-        num_obj = len(image)
-        loss = 0
-        for i in range(num_obj):
-            # TODO(Jiyong): need to fix error("ValueError: only one element
-            # tensors can be converted to Python scalars") for trying
-            # to(device) at above.
-            _image = image[i].to(device)
-            _truth_orient = th.squeeze(truth_orient[i]).to(device)
-            _truth_dim = th.squeeze(truth_dim[i]).to(device)
-
-            dim, quat = model(_image)
-
-            scale_loss = scale_loss_func(dim, _truth_dim)
-            orient_loss = orientation_loss(quat, _truth_orient)
-
-            # orient_loss = orientation_loss(orient, truth_orient, truth_conf)
-            # truth_conf = th.max(truth_conf, dim=1)[1]
-            # conf_loss = conf_loss_func(conf, truth_conf)
-            # loss_theta = conf_loss + opts.w * orient_loss
-            # loss = opts.alpha * scale_loss + loss_theta
-
-            loss += opts.alpha * scale_loss + orient_loss
-
-        return loss / num_obj
+        # TODO(Jiyong): chage to collate_fn
+        image = data[Schema.CROPPED_IMAGE].to(device)
+        # image = image * (1.0/255.0) - 0.5
+        _, _, c, h, w = image.shape
+        image = image.view(-1, c, h, w)
+        truth_orient = data[Schema.ORIENTATION].to(device)
+        truth_orient = truth_orient.view(-1,4)
+        truth_dim = data[Schema.SCALE].to(device)
+        truth_dim = truth_dim.view(-1,3)
+
+        # dim, quat = model(image)
+        # scale_loss = scale_loss_func(dim, truth_dim)
+        # orient_loss = orientation_loss_func(quat, truth_orient)
+        # loss = opts.alpha * scale_loss + orient_loss
+        
+        dim, quat = model(image)
+        print('...')
```

Comment:
Reminder to remove these print statements before merging.

Reply:
I will remove these part before merging.